### PR TITLE
Update: Overrides for space-unary-ops (fixes #5060)

### LINKED ISSUE #5060: Rule enhancement: More granular control in space-unary-ops
### Current state

This rule allows us to configure whether we expect spaces between unary operators for either word operators (`delete`, `void`, `typeof` etc.) and non-word (`+`, `-`, `++`, `!`, etc.).

### Problem

Some might prefer different spacing rules for individual non-word operators. For me personally, I prefer to have the negation operator `!` always separated from its operand, but I never separate `++` or `--` from its operand.

### Proposal

Extend the rule's configuration so that we can configure the spacing rules for individual operators. One example might be to add an object as third argument where the object's key would be the operator and the key's value would be the rule to apply:

```js
{
  'space-unary-ops': [1, {
    words: false,
    nonwords: false,
    '!': true  // Overrides spacing rule for ! to always have space
  }]
}
```

> The configuration scheme above is just an idea. The goal is to allow more fine-grained control over where the spaces are expected and where they are not.

Thanks!

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/30177095-rule-enhancement-more-granular-control-in-space-unary-ops?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by afahim**
In the process of implementing this, I've abstracted some of the logic for handling spaces out into their own functions, so as to make the logic easier to reason about.  Would love to hear your thoughts.

### CODE REVIEW

File: docs/rules/space-unary-ops.md [hunk 1]
```diff
@@ -39,10 +39,26 @@ qux = !!baz;
...
+
+```js
+    "space-unary-ops": [
+        2, {
+          "words": true,
+          "nonwords": false,
+          "overrides": {
+            "new": false,
+            "++": true
+    }}]
```
Comment (nzakas) [inline docs/rules/space-unary-ops.md]: Can you pull the closing brace for the overrides object up one line?

**Comment by nzakas** [comment]
Looks fine to me so far.

File: docs/rules/space-unary-ops.md (hunk 1 above)
Comment (afahim) [inline docs/rules/space-unary-ops.md]: @nzakas I've updated it. Let me know if there is anything else.

**Comment by nzakas** [comment]
@afahim just comment back here when you're ready for a last review.

**Comment by afahim** [comment]
@nzakas I believe this is ready for review, I've made the formatting change that you suggested.

**Comment by nzakas** [comment]
Lgtm. Thanks!

**Pull request merged**